experimental/InvariantTheory/src/InvariantTheory.jl

```diff
@@ -0,0 +1,523 @@
+    @assert group(X) == group(Y)
+    G = group(X)
+    R = base_ring(group_ideal(G))
+    Mat = block_diagonal_matrix(R, [Matrix(representation_matrix(X)), Matrix(representation_matrix(Y))])
+    return RepresentationReductiveGroup(G, Mat)
+end
+
+function direct_sum(V::Vector{RepresentationReductiveGroup})
+    n = length(V)
+    G = group(V[1])
+    for i in 2:n
+        @assert G == group(V[i])
+    end
+    R = base_ring(group_ideal(G))
+    Mat = block_diagonal_matrix(R, [Matrix(representation_matrix(V[i])) for i in 1:n])
+    return RepresentationReductiveGroup(G, Mat)
+end
+
+function tensor(X::RepresentationReductiveGroup, Y::RepresentationReductiveGroup)
+    @assert group(X) == group(Y)
+    Mat = kronecker_product(representation_matrix(X), representation_matrix(Y))
+    return RepresentationReductiveGroup(group(X), Mat)
+end
+
+function tensor(V::Vector{RepresentationReductiveGroup})
+    n = length(V)
+    for i in 2:n
+        @assert group(V[1]) == group(V[i])
+    end
+    Mat = representation_matrix(V[1])
+    for i in 2:n
+        Mat = kronecker_product(Mat,representation_matrix(V[i]))
+    end
+    return RepresentationReductiveGroup(group(V[1]), Mat)
+end
+
+###############
+
+#computes the representation matrices of SL_m acting over m-forms of symmetric degree sym_deg
+function rep_mat_(G::ReductiveGroup, sym_deg::Int)
+    G.group[1] == :SL || error("Only implemented for SLm")
+    m = G.group[2]
+    R = base_ring(G.group_ideal) # TODO: probably should have a getter function for this
+    mixed_ring, t = polynomial_ring(R, "t" => 1:m)
+    group_mat = natural_representation(G)
+    new_vars = group_mat*t
+    
+    b = degree_basis(mixed_ring,sym_deg)
+    n = length(b)
+    
+    # transform the b elements
+    images_of_b = [evaluate(f, new_vars) for f in b]
+
+    mat = zero_matrix(R, n, n)
+    for j in 1:n
+        f = images_of_b[j]
+        x = mixed_ring()
+        # express f as a linear combination of degree_basis
+        for i in 1:n
+            c = coeff(f, leading_exponent(b[i]))
+            mat[i,j] = c / leading_coefficient(b[i])
+        end
+    end
+    return mat
+end
+
+#computes symmetric degree basis (of the first m variables) WITH multinomial coefficients!
+function degree_basis(R::MPolyRing, t::Int)
+    m = ngens(R)
+    C = zero_matrix(ZZ, m, m)
+    for i in 1:m
+      C[i,i] = -1 
+    end
+    d = zeros(m)
+    A = ones(m)
+    b = [t]
+    P = polyhedron((C, d), (A, b))
+    L = lattice_points(P)
+    W = Vector{MPolyRingElem}(undef,0)
+    for l in L
+        v = R(1)
+        for i in 1:m
+            v = v*gen(R,i)^l[i]
+        end
+        v = v*multinomial(t,l)
+        push!(W,v)
+    end
+    #we reverse here to get the natural order of a degree basis, eg x^2, 2xy, y^2.
+    return reverse(W)
+end
+
+#used to compute multinomial expansion coefficients (used in degree_basis)
+function multinomial(n::Int, v::Union{Vector{Int64},PointVector{ZZRingElem}})
```

Comment:
Let's make it a bit more flexible (that was helpful for testing for me)
```suggestion
function multinomial(n::Int, v::Union{Vector{T},PointVector{T}}) where T <: IntegerUnion
```